packages/fiori-freestyle-writer/test/__snapshots__/listdetail.test.ts.snap

```diff
@@ -2050,175 +2050,170 @@ errorText=Sorry, a technical error occurred! Please try again later.",
-              if (moduleDefinition && moduleDefinition.dependencies) {
-                moduleDefinition.dependencies.forEach(function (dependency) {
-                  if (dependency.url && dependency.url.length > 0 && dependency.type === \\"UI5LIB\\") {
-                    jQuery.sap.log.info(
-                      \\"Registering Library \\" +
-                      dependency.componentId +
-                      \\" from server \\" +
-                      dependency.url
-                    );
-                    jQuery.sap.registerModulePath(dependency.componentId, dependency.url);
-                  }
+        return result;
+    };
+
+    const fioriToolsGetManifestLibs = function (manifestPath) {
+        const url = manifestPath;
+        return new Promise(function (resolve, reject) {
+            $.ajax(url)
+                .done(function (manifest) {
+                    resolve(updateManifest(manifest));
+                })
+                .fail(function () {
+                    reject(new Error(\\"Could not fetch manifest at '\\" + manifestPath));
                 });
-              }
+        });
+    };
+
+    const updateModuleDefinition = function (data) {
+        if (data) {
+            Object.keys(data).forEach(function (moduleDefinitionKey) {
+                const moduleDefinition = data[moduleDefinitionKey];
+                if (moduleDefinition && moduleDefinition.dependencies) {
+                    moduleDefinition.dependencies.forEach(function (dependency) {
+                        if (dependency.url && dependency.url.length > 0 && dependency.type === \\"UI5LIB\\") {
+                            jQuery.sap.log.info(
+                                \\"Registering Library \\" + dependency.componentId + \\" from server \\" + dependency.url
+                            );
+                            jQuery.sap.registerModulePath(dependency.componentId, dependency.url);
+                        }
+                    });
+                }
             });
-          }
+        }
+    };
+
+    /**
+     * Registers the module paths for dependencies of the given component.
+     * @param {string} manifestPath The the path to the app manifest path
+     * for which the dependencies should be registered.
+     * @returns {Promise} A promise which is resolved when the ajax request for
+     * the app-index was successful and the module paths were registered.
+     */
+    sap.registerComponentDependencyPaths = function (manifestPath) {
+        /*eslint-disable semi, consistent-return*/
+        return fioriToolsGetManifestLibs(manifestPath).then(function (libs) {
+            if (libs && libs.length > 0) {
+                let url = \\"/sap/bc/ui2/app_index/ui5_app_info?id=\\" + libs;
+                const sapClient = jQuery.sap.getUriParameters().get(\\"sap-client\\");
+                if (sapClient && sapClient.length === 3) {
+                    url = url + \\"&sap-client=\\" + sapClient;
```

Comment:
I think this will break the generated URL as now it would include "" but the url needs to be in the form:
`<host><path>&sap-client=123`